core/state_transition.go

```diff
@@ -363,14 +375,18 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 		// Increment the nonce for the next transaction
 		st.state.SetNonce(msg.From(), st.state.GetNonce(sender.Address())+1)
 		ret, st.gas, vmerr = st.evm.Call(sender, st.to(), st.data, st.gas, st.value)
-		if vmerr == nil && *msg.To() == GetStateConnectorContract(chainID, timestamp) && len(st.data) >= 36 && len(ret) == 32 {
-			if GetStateConnectorActivated(chainID, timestamp) &&
-				bytes.Equal(st.data[0:4], SubmitAttestationSelector(chainID, timestamp)) &&
-				binary.BigEndian.Uint64(ret[24:32]) > 0 {
-				err = st.FinalisePreviousRound(chainID, timestamp, st.data[4:36])
-				if err != nil {
-					log.Warn("Error finalising state connector round", "error", err)
-				}
+		if vmerr == nil &&
+			*msg.To() == stateConnectorContract(chainID, timestamp) &&
+			len(st.data) >= 36 &&
+			len(ret) == 32 &&
+			isStateConnectorActivated(chainID, timestamp) &&
+			// 4 first bytes corresponds to the function name
+			bytes.Equal(st.data[0:4], submitAttestationSelector(chainID, timestamp)) &&
+			// determine whether this is read-only call
+			binary.BigEndian.Uint64(ret[24:32]) > 0 {
+			err = st.stateConnector.finalisePreviousRound(chainID, timestamp, st.data[4:36])
```

Comment:
Lines 379 to 386 and line 387 should not be at  the same level of indentation IMO, it's confusing 🤔 Was this code go-fmted?

Reply:
Actually, since it is indeed formatted, I have an alternate suggestion @bogdanprodanj: Put this logic in an appropriately named function.

```go
func shouldFinalize(...) bool {
	if *msg.To() != stateConnectorContract(chainID, timestamp) {
		return false
	}
	if len(st.data) < 36 {
		return false
	}
	if len(ret) != 32 {
		return false
	}
	if !isStateConnectorActivated(chainID, timestamp) {
		return false
	}
	if !bytes.Equal(st.data[0:4], submitAttestationSelector(chainID, timestamp)) {
		return flase
	}
	if binary.BigEndian.Uint64(ret[24:32]) <= 0 {
		return false
	}

	return true
}
```

The only issue with that is that it needs access to the following variables: `chainID`, `timestamp`, `msg`, `st` and `ret`.